scripts/generate_crc_catalog.py

```
@@ -0,0 +1,646 @@
+def _update_existing_entry(existing: dict, incoming: dict) -> None:
+    """Updates missing metadata and checksums on an existing catalog entry."""
+    if incoming.get("cdnUrl"):
+        existing["cdnUrl"] = incoming["cdnUrl"]
+    for key in ("exeCrc", "iniCrc"):
+        if not existing.get(key) and incoming.get(key):
+            existing[key] = normalize_hex(incoming[key])
+    if not existing.get("sha256") and incoming.get("sha256"):
+        existing["sha256"] = incoming["sha256"]
+
+
+def merge_catalogs(existing: list[dict], crawled: list[dict]) -> list[dict]:
```

Ruff (0.16.3)

[warning] Too many branches (15 > 12)

(PLR0912)